### PR TITLE
issue/25 Allow external blocks

### LINKED ISSUE #25: Allow external blocks
* Associate external blocks with branching article at `models` and `branchedModels` 
https://github.com/adaptlearning/adapt-contrib-branching/blob/c9f7ab9ba140949b4ed031118eaf44f108356173/js/BranchingSet.js#L167-L181
* Ensure save/restore works effectively
* Single branching association only

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #25 
fixes https://github.com/adaptlearning/adapt_framework/issues/2799

merged #23, #26, #28

External blocks can be added to a branching article. By default, they will be available in their original location but will also be included at the end of the specified branching set.
Note: External blocks added to a branching article will not retain their own `_attemptStates` history, this is as their history is used to save/restore the branching scenario.

### Added
* `_containerId` to allow blocks to be assigned to a branching article id

### Removed
* Redundant behaviour, swapping from `Adapt.offlineStorage` etc to `offlineStorage` with direct module imports
* Redundant `trackingPosition.js`, as these functions are now in core

Requires https://github.com/adaptlearning/adapt-contrib-core/pull/118

### CODE REVIEW

File: js/adapt-contrib-branching.js [hunk 1]
```diff
@@ -23,12 +25,23 @@ class Branching extends Backbone.Controller {
...
+    this.warnForSpoorMisconfiguration();
+    this._isAwaitingDataReady = false;
     this.setupBranchingModels();
     this.setupEventListeners();
     Adapt.trigger('branching:dataReady');
   }
 
+  warnForSpoorMisconfiguration() {
+    const config = Adapt.config.get('_spoor');
+    const isMisConfigured = (config?._isEnabled && config?._tracking?._shouldStoreAttempts === false);
```
Comment (cahirodoherty-learningpool) [inline js/adapt-contrib-branching.js:40]:
```suggestion
    const isMisconfigured = (config?._isEnabled && config?._tracking?._shouldStoreAttempts === false);
```

(Minor typo)

File: js/adapt-contrib-branching.js [hunk 2]
```diff
@@ -23,12 +25,23 @@ class Branching extends Backbone.Controller {
...
+    if (!isMisConfigured) return;
```
Comment (cahirodoherty-learningpool) [inline js/adapt-contrib-branching.js:41]:
```suggestion
    if (!isMisconfigured) return;
```

(Minor typo)

Comment (oliverfoster) [reply js/adapt-contrib-branching.js]: You can do multi-line suggestions, if you use shift to select and then click the comment plus at the bottom of the selection.

File: js/adapt-contrib-branching.js [hunk 3]
```diff
@@ -146,7 +171,7 @@ class Branching extends Backbone.Controller {
 
   contineAfterBranchChild(model) {
```
Comment (cahirodoherty-learningpool) [inline js/adapt-contrib-branching.js:172]: Typo here, but it's not your work. Not sure where it gets called

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by eleanor-heath — APPROVED** [review]

**Pull request merged**